Starting up

EduStormCrow opened the pull request:
This PR is for setting the fundation for the development of the Notes app

EduStormCrow committed:
Starting up

mzorz reviewed (commented):
Hi there! Nice to see you starting from scratch 🎉. Left a few comments in the code review 👍

mzorz commented:
Left a few comments in the code review, also some recommendations for a start:

1 - the PR should have a description, even if a super short one, that describes what it is about.

2 - I'd expect a `README.md` file to exist in the root folder, and a brief explanation of what the project is about. Please add it to your working branch (`start-branch`) and update the remote repo.
There's actually a README.md file in `master`. Your branch should have been branched off from `master` as well. It's ok as this is a first setup.

3 - the notes list is empty at first run, it'd be nice if it showed an empty placeholder or prompt the user to start writing notes, like "Hey! Make your first note bu clicking the + button".

4 - when adding a note, the keyboard goes up and hides the "save" FAB. It took me a fraction of a second to realize, but that might be frustrating for users. You might want to try and think how the FAB can be moved up when the keyboard is up if possible at all, or make the keyboard's default action for the second (last) EditText to be "SAVE" instead of "ENTER".
![edu2](https://user-images.githubusercontent.com/6597771/42941694-edb73694-8b33-11e8-9ec2-14faa664ce93.gif)


Other than that, this looks pretty cool Edu! 😄 I'm excited about the work you've done. Keep it on! 💪

EduStormCrow committed:
Adding gitignore

Based on codereview, adding a more descriptive gitignore

EduStormCrow commented on .gitignore:
```diff
@@ -0,0 +1,10 @@
+.DS_Store
+/build
+/captures
+.externalNativeBuild
```
thanks for the reference, I have added the default one and then will be keeping up to date based on future libraries usage.

mzorz replied:
for the sake of completeness, can you tell me here which's  the commit  hash that addresses it?

mzorz replied:
ah - found it https://github.com/EduStormCrow/Notes/pull/1/commits/9c88e53ec35ecb19e1858d2b85f8526b8a2b3844

EduStormCrow commented on app/build.gradle:
```diff
@@ -0,0 +1,46 @@
+    defaultConfig {
+        applicationId "com.darkwoods.notes"
+        minSdkVersion 21
+        targetSdkVersion 27
```
Thanks for the additional info! Changing it to 28

EduStormCrow committed:
Updating Sdk to 28 and removing unnecesary info

Added the CompiledSdkVersion to 28  and removed the libraries/plugins that weren't in use

EduStormCrow commented on app/build.gradle:
```diff
@@ -0,0 +1,46 @@
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation"org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
```
My idea was to use it on start, but got deprioritized. So I will be removing this from now... (commit #102f366)

EduStormCrow commented on app/build.gradle:
```diff
@@ -0,0 +1,46 @@
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    // CardView
+    compile 'com.android.support:cardview-v7:23.3.+'
```
Slipped when Copy+Pasting from android documentation... already fixed this on (commit #102f366)

EduStormCrow commented on app/src/androidTest/java/com/darkwoods/notes/ExampleInstrumentedTest.kt:
```diff
@@ -0,0 +1,24 @@
+//        val appContext = InstrumentationRegistry.getTargetContext()
+//        assertEquals("com.darkwoods.notes", appContext.packageName)
+//    }
+}
```
Seems good... will keep that in mind.

EduStormCrow committed:
removing

removing this until we address all the testing suite

EduStormCrow committed:
Class visual Cleanup

Removing spaces and translating comments to english

EduStormCrow committed:
Visual cleanup

removing spaces and tabs

EduStormCrow committed:
Removing commented code

EduStormCrow committed:
Visual Cleanup

removed extra lines

EduStormCrow committed:
Removed commented code

EduStormCrow committed:
Renamed FAB to addNoteFAB

EduStormCrow commented on app/src/main/res/layout/activity_list_notes.xml:
```diff
@@ -0,0 +1,26 @@
+
+    <!-- Floating action button that launches AddTaskActivity when clicked -->
+    <android.support.design.widget.FloatingActionButton
+        android:id="@+id/fab"
```
Done (commit #5a21eca)

EduStormCrow committed:
Renamed method parameter with the m prefix

EduStormCrow commented on app/src/main/java/com/darkwoods/notes/ListNotesActivity.java:
```diff
@@ -0,0 +1,238 @@
+
+
+    @Override
+    public void onLongClickListener(Note mNote) {
```
Fixed. Thanks for the Rename tip, I have been using it for a while now... good way of not messing things up.

EduStormCrow committed:
Removed unnecesary spacing and lines

EduStormCrow commented on app/src/main/java/com/darkwoods/notes/NotesAdapter.java:
```diff
@@ -0,0 +1,188 @@
+    public Note getNoteOnPosition(int pos){
+        return mNotesList.get(pos);
+    }
+
```
fixed in #8630c2

EduStormCrow commented on app/src/main/java/com/darkwoods/notes/NotesAdapter.java:
```diff
@@ -0,0 +1,188 @@
+
+        void onLongClickListener(Note mNote);
+    }
+
```
fixed in #8630c2

EduStormCrow committed:
Removed unnecessary spacing and lines

EduStormCrow commented on app/src/main/java/com/darkwoods/notes/database/NoteVModel.java:
```diff
@@ -0,0 +1,23 @@
+public class NoteVModel extends ViewModel {
+
+    private LiveData<Note> noteLiveData;
+
```
fixed #8630c2

EduStormCrow committed:
Removed extra tabs

EduStormCrow commented on app/src/main/java/com/darkwoods/notes/database/NoteVModel.java:
```diff
@@ -0,0 +1,23 @@
+    private LiveData<Note> noteLiveData;
+
+
+        public NoteVModel(AppDatabase database, int noteID) {
```
fixed #cd4db64

EduStormCrow commented:
Thanks @mzorz  for doing this... I fixed part of the code and added a few comments to your reviews.

Based on your questions...
1) Added the description.
2) I added a readme in the master to had something to compere while doing the merge. I will update that one with a basic description.
3) Yes, that was the first item of the TODO list. I will add that functionality later on.
4) Yes, that was the 3rd item of the TODO list. I will work on that, I believe moving the button will be better as you are adding a description that could have multiple lines, so the enter approach will not be intuitive.
Again, thanks for taking a look at my code!

EduStormCrow commented on app/src/main/java/com/darkwoods/notes/ListNotesActivity.java:
```diff
@@ -0,0 +1,238 @@
+    /**
+     * RecyclerView item decoration - give equal margin around grid item
+     */
+    public class GridSpacingItemDecoration extends RecyclerView.ItemDecoration {
```
I need to find where I took it from, some comment in stackoverflow.com

mzorz commented on app/src/main/java/com/darkwoods/notes/database/Note.java:
```diff
@@ -0,0 +1,65 @@
+
+    @PrimaryKey(autoGenerate = true)
+    int id;
+    private String title;
```
Good choice! that explains it. Thanks for clarifying.

mzorz commented:
Excellent! Thanks for fixing and taking the exposed considerations into account. 
LGTM :shipit:
